scripts/webpack-utils/BuildCompiler.js

```diff
@@ -0,0 +1,233 @@
+'use strict';
+
+/**
+ * A helper class for the webpack build compiler (as opposed to the `watch`
+ * compiler).
+ *
+ * @see - https://webpack.js.org/api/node/
+ * @see - https://webpack.js.org/api/compiler-hooks
+ *
+ * The primary purpose of this helper class is to print useful messages about
+ * the build status.  While the compiler is running (building), a status message
+ * will be logged to the console every second.  When the build finishes, errors
+ * and warnings will be more clearly displayed, rather than being hidden amongst
+ * the rest of the webpack stats.
+ */
+
+const chalk = require('chalk');
+const webpack = require('webpack');
+
+module.exports = class BuildCompiler {
+  /**
+   * @see - this.constructor
+   */
+  static factory (name, webpackConfigs) {
+    return new BuildCompiler(name, webpackConfigs);
+  }
+
+  /**
+   * @constructor
+   * @private
+   *
+   * Create a new BuildCompiler, and register the default webpack compiler
+   * hooks.
+   *
+   * @param {string} name
+   *   The name of the compile job (arbitrary), used for logging
+   * @param {Array} webpackConfigs
+   *   An array of webpack config objects - @see - https://webpack.js.org/configuration/#options
+   *
+   * @returns {BuildCompiler}
+   *   A new BuildCompiler instance
+   */
+  constructor (name, webpackConfigs = []) {
+    this.name = name;
+    this.compiler = webpack(webpackConfigs);
+
+    this.clearTimer();
+    this.initialize();
+  }
+
+  /**
+   * @private
+   *
+   * Stop the internal compile timer.
+   *
+   * @returns {void}
+   */
+  clearTimer () {
+    if (this.timer) {
+      clearInterval(this.timer);
+    }
+
+    this.timer = null;
+    this.secondsCompiling = 0;
+  }
+
+  /**
+   * @private
+   *
+   * Start the internal compile timer.  A message will be logged to the console
+   * every second until the timer is cleared.
+   *
+   * @returns {void}
+   */
+  setTimer () {
+    // @todo - what should happen when this.timer already exists?
```

Comment:
@brainthinks Suggestion for todo:

If `timer` isn't null then just leave the method (early return)

Reply:
i could, but we don't currently have a use case for it, so i left it as a todo.  I'll make the change, though

Reply:
done